virtcontainers/container.go

```diff
@@ -1142,6 +1142,11 @@ func (c *Container) detachDevices() error {
 		}
 	}
 
+	for _, dev := range c.devices {
+		// ignore the device not exist error
+		c.sandbox.devManager.RemoveDevice(dev.ID)
```

Comment:
ignore it, but at least log the error please

Reply:
Yes, please log the error.

Reply:
@WeiZhang555 I think we should identify one type of error that we can ignore on purpose (and log something), so that we don't ignore other errors.

Also, is there any reason why we cannot call this `RemoveDevice()` function right after we detached the device with `DetachDevice()` ? This would prevent from running twice through the same list of devices.

Reply:
Will log this.

@sboeuf We can, let me modify it a bit.

Reply:
And `ErrDeviceNotExist` is the only error RemoveDevice can return, but I will add a judgement for future extension.